src/fetch_satellite_singlepass.py

```diff
@@ -0,0 +1,96 @@
+import datetime
+import pandas as pd
+import imaplib
+import json
+import os
+import sys
+import email
+from pathlib import Path
+import mimetypes
+import logging
+_log = logging.getLogger(__name__)
+
+root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+sys.path.insert(0, root_dir)
+
+# keep variable names/structure from the first script
+rough_data = Path(root_dir) / "data" / "sat_singlepass"
+if not rough_data.exists():
+    rough_data.mkdir(parents=True)
+
+
+mail_alarms_json = Path("mail_alarms.json")
+
+
+
+def fetch_metop_sst_attachments():
+    secrets_file = Path(root_dir) / "email_secrets.json"
+    if not secrets_file.exists():
+        _log.error(f"Did not find secrets file {secrets_file}. Cannot read emails. Skipping")
+        return
+    with open(f"{root_dir}/email_secrets.json", encoding="utf-8") as json_file:
+        secrets = json.load(json_file)
+
+    # connect to gmail and search for specific MetOp SST mails
+    mail = imaplib.IMAP4_SSL("imap.gmail.com")
+    try:
+        mail.login(secrets["email_username"], secrets["email_password"])
+    except imaplib.IMAP4.error:
+        _log.error(f"Incorrect credentials in {secrets_file}. Skipping email read")
+        return
+
+    mail.select("inbox")
+    result, data = mail.search(None, '(FROM "teradm@linfernp31.srx.bsh.de" SUBJECT "MetOp SST")')
```

Comment:
Is this FROM search going to work with a forwarded mail? maybe just the subject search is specific enough?